internetarchive/utils.py

```diff
@@ -332,7 +334,7 @@ def remove_none(obj):
         return obj
 
 
-def delete_items_from_dict(d, to_delete):
+def delete_items_from_dict(d: dict, to_delete):
```

Comment:
A complicated case. For now this seems fitting but I think the whole function should be refactored.

`Dict | List` seems to be a fitting return type for now.
`d` would also have to be `Dict | List`, `Mapping | Iterable` would be preferred, but since the types are checked explicitly in this case, I would stick with `Dict | List` for now.

Reply:
`pyupgrade --py310-plus` will remove the `annotations` import  and then lowercase both `dict` and `list` so let’s go that way too.

The function name pushes me too keep the argument type as `dict`.  Can items be deleted from a Mapping or would we need to specify as MutableMappling?

Reply:
Yeah, it'd have to be `MutableMapping` for the `__delitem__` method.

Reply:
It would have to be a `MutableMapping`.

But I would use `dict | list` for now, since there are those explicit type checks anyways.

Reply:
Especially since you also cannot delete items from iterables. I'm not sure what the appropriate base class for those would be. There probably is none since the iterable interface is rather basic and it would not make much sense to have a common mutation method.